src/services/waku.ts

```diff
@@ -192,3 +192,33 @@ export const useLatestTopicData = <Data>(
 
 	return { ...state, lastUpdate, data, payload }
 }
+
+export const fetchLatestTopicData = <Msg extends Message>(
+	waku: WakuLight,
+	decoders: Decoder<Msg>[],
+	callback: (message: Promise<Msg | undefined>) => Promise<boolean | void>,
+	options?: QueryOptions | undefined,
+	watch = false
+) => {
+	// eslint-disable-next-line @typescript-eslint/no-extra-semi
+	;(async () => {
+		const generator = waku.store.queryGenerator(decoders, {
+			pageDirection: PageDirection.BACKWARD,
+			pageSize: 1,
+		})
+
+		for await (const messages of generator) {
+			for (const message of messages) {
+				if (await callback(message)) {
+					return
+				}
+			}
+		}
```

Comment:
How heavy can this be?

Reply:
So, `messages` should only ever contain one element (I did it in a loop to make sure we're not missing anything if we forget to set the page size for example), and the `callback` is just decoding the message. If it's successful, then it just breaks the loop. In most cases it should just be `O(1)` because the first message should be valid.

Basically it fetches the latest message from Waku, check its validity, and move on to the next only if it's invalid.